### ISSUE TITLE
reaper freezes and can't kill it

### ISSUE THREAD

**Issue by cjunekim** [eiz/SynchronousAudioRouter#20]
I am happy with sar overall. But sometimes the reaper process gets stuck, usually when the preference window is open.

The bigger issue is I can't terminate the process and I can't even restart the OS, or turn off the computer.

I tried taskkill /t /f with admin privillege and it failed to kill reaper.

**Comment by eiz** [comment]
Yeah, I've experienced this on occasion. I believe there is a deadlock while closing the SAR control device that can occur. Consistently happens for me if I hard kill reaper through task manager, then run it again.

**Comment by pannal** [comment]
I've experienced the same, but with [the latest pre-release](https://github.com/eiz/SynchronousAudioRouter/issues/13#issuecomment-294436985) at least the hard lockup doesn't happen anymore.

**Comment by cjunekim** [comment]
I've been already using the latest pre-relase(20170417) and it happens often.

**Comment by pannal** [comment]
That usually happens when I open the preferences window and another application uses the sound card. That also almost always results in a REAPER crash (version v5.50rc20/x64). Whenever I want to play around with preferences I now close any application using the virtual sound cards I expose via ASIO4ALL and SAR via REAPER.
